src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -221,6 +223,10 @@ def setUp(self):
                     "Test requires {} {}; {} specified".format(
                         expected_count, host_type, actual_count))
 
+        #Storage setup if requested in test input file
+        if self.nvme_parameter == "nvme":
+            server_utils.storage_prepare(self.hostlist_servers)
```

Comment:
Is there error handling that needs to be wrapped around this call?

Reply:
> After looking at the code not sure if we want to add try block. Avocado setup function is going to catch this as ERROR and raise the "TestSetupFail: Error preparing NVMe storage".

That's right.  But we don't want Avocado tests to **ERROR**, we want them to **FAIL**.  Those two conditions report differently in Jenkins where the former means to Jenkins that running the test framework failed in some way whereas the latter means that a test failed.  The former means users have to look through the log file to figure out why Avocado failed but the latter means they look in the Test result to see why the test failed.  Clearly this is a case of the latter.

> Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:576
> Traceback (most recent call last):
> File "/home/samirrav/daos/src/tests/ftest/pool/attribute.py", line 84, in setUp
> super(PoolAttributeTest, self).setUp()
> File "/home/samirrav/daos/src/tests/ftest/util/apricot/apricot/test.py", line 229, in setUp
> server_utils.storage_prepare(self.hostlist_servers)
> File "/home/samirrav/daos/src/tests/ftest/util/server_utils.py", line 351, in storage_prepare
> raise ServerFailed("Error preparing NVMe storage")
> ServerFailed: Error preparing NVMe storage

If this operation is wrapped in a `try:`/`except:` block to catch the `ServerFailed` exception and then call `self.fail()`, this should make the Avocado **ERROR** a **FAIL** instead.

Reply:
ok, so it will be the same as other comments. I will wait for your investigation.